src/hooks/useAppRouter.ts

```diff
@@ -55,16 +57,23 @@ export function useAppRouter() {
     const storedNetworkAlias = getNetworkName(Number(networkIdxStore));
     const networkParam = castNetworkName(network.value);
 
+    const getRandomizedEndpoint = (): string => {
+      const endpointsWithoutLightClient = selectedChain.endpoints.filter(
+        (it) => !checkIsLightClient(it.endpoint)
+      );
+      return getRandomFromArray(endpointsWithoutLightClient).endpoint;
+    };
+
     const isReload =
       networkIdxStore === null ||
       networkIdxStore === undefined ||
       storedNetworkAlias !== networkParam;
     if (isReload) {
-      // Memo: `defaultCurrency` and `selectedEndpoint` are updated when the app connects to the API
-      const isFirstTimeVisitor = localStorage.length === 2;
+      // Memo: `defaultCurrency`, `isAppliedRandomEndpoint` and `selectedEndpoint` are updated when the app connects to the API
+      const isFirstTimeVisitor = localStorage.length === 3;
```

Comment:
why don't we check each key of local storage? If we manipulate local storage elsewhere, there is a high possibility of a bug.

Reply:
@sirius651  Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/820/commits/34bba494effeb449c47f84856b563c6a7ea6b9f2) commit.